src/Sentry.Unity/WebGL/SentryWebGL.cs

```diff
@@ -0,0 +1,157 @@
+            // Additionally, we could recognize the situation when the unity gets stuck due to an error in JS/native:
+            //    "An abnormal situation has occurred: the PlayerLoop internal function has been called recursively.
+            //     Please contact Customer Support with a sample project so that we can reproduce the problem and troubleshoot it."
+            // Maybe we could write a file when this error occurs and recognize it on the next start. Like unity-native.
+            options.CrashedLastRun = () => false;
+
+            // Disable async when accessing files (e.g. FileStream(useAsync: true)) because it throws on WebGL.
+            options.UseAsyncFileIO = false;
+        }
+    }
+
+    internal class WebBackgroundWorker : IBackgroundWorker
+    {
+        private readonly SentryMonoBehaviour _behaviour;
+        private readonly UnityWebRequestTransport _transport;
+
+        public WebBackgroundWorker(SentryUnityOptions options, SentryMonoBehaviour behaviour)
+        {
+            _behaviour = behaviour;
+            _transport = new UnityWebRequestTransport(options);
+        }
+
+        public bool EnqueueEnvelope(Envelope envelope)
+        {
+            _ = _behaviour.StartCoroutine(_transport.SendEnvelopeAsync(envelope));
+            return true;
+        }
+
+        public Task FlushAsync(TimeSpan timeout) => Task.CompletedTask;
+
+        public int QueuedItems { get; }
+    }
+
+    internal class UnityWebRequestTransport : HttpTransportBase
+    {
+        private readonly SentryUnityOptions _options;
+
+        public UnityWebRequestTransport(SentryUnityOptions options)
+            : base(options)
+        {
+            _options = options;
+        }
+
+        // adapted HttpTransport.SendEnvelopeAsync()
+        internal IEnumerator SendEnvelopeAsync(Envelope envelope)
+        {
+            using var processedEnvelope = ProcessEnvelope(envelope);
+            if (processedEnvelope.Items.Count > 0)
+            {
+                // Send envelope to ingress
+                var httpRequest = CreateRequest(processedEnvelope);
+                var www = CreateWebRequest(httpRequest);
+                yield return www.SendWebRequest();
+
+                var response = GetResponse(www);
+                if (response != null)
+                {
+                    HandleResponse(response, processedEnvelope);
+                }
+            }
+        }
+
+        private UnityWebRequest CreateWebRequest(HttpRequestMessage message)
+        {
+            using var contentStream = ReadStreamFromHttpContent(message.Content);
+            var www = new UnityWebRequest
+            {
+                url = message.RequestUri.ToString(),
+                method = message.Method.Method.ToUpperInvariant(),
+                uploadHandler = new UploadHandlerRaw(((MemoryStream)contentStream).ToArray()),
```

Comment:
Is this cast going to be safe straight up?

https://github.com/getsentry/sentry-dotnet/blob/f28b1c513547413340c5077189b57b051e15fdf9/src/Sentry/Http/HttpTransportBase.cs#L228